wagtail_newsletter/panels.py

```diff
@@ -1,6 +1,54 @@
+import logging
+
+from django.core.exceptions import ImproperlyConfigured
+from django.utils.html import format_html
 from wagtail.admin.panels import Panel
 
+from . import campaign_backends, models
+
+
+logger = logging.getLogger(__name__)
+
 
 class NewsletterPanel(Panel):
     class BoundPanel(Panel.BoundPanel):
         template_name = "wagtail_newsletter/panels/newsletter_panel.html"
+
+        instance: "models.NewsletterPageMixin"
+
+        def get_context_data(self, parent_context=None):
+            context = super().get_context_data(parent_context) or {}
+            backend = campaign_backends.get_backend()
+            campaign = None
+
+            if self.instance.pk:
+                if self.instance.newsletter_campaign:
+                    try:
```

Comment:
Being extra careful here: this panel is rendered whenever the page editor view is loaded, so we don't want any errors breaking the regular page editing functionality.